src/stdlib_stats_moment_all.fypp

```diff
@@ -61,10 +64,11 @@ contains
         n = real(size(x, kind = int64), dp)
 
         if (present(center)) then
-         res = sum((real(x, dp) - center)**order) / n
+          center_ = center
         else
-         res = sum((real(x, dp) - mean(x))**order) / n
+          center_ = mean(x)
         end if
```

Comment:
Could this not also be replaced by `optval`?

```suggestion
        center_ = optval(center, mean(x))
```

and likewise in two other instances below?

Reply:
Is it ok for efficiency? I mean: would `mean(x)` not be evaluated, even if not needed?